### ISSUE TITLE
httpx Parse panic

### ISSUE THREAD

**Issue by wuqinqiang** [zeromicro/go-zero#3838]
**Describe the bug**

**To Reproduce**
Steps to reproduce the behavior, if applicable:

1. The code is
- 

2. The error is

The following logs

   
**Expected behavior**

**Screenshots**

**Environments (please complete the following information):**
 - OS: [e.g. Linux]
 - go-zero v1.5.6
 - goctl 1.5.4 

**More description**

request struct：

```go
type ExampleReq {
	Numbers  []int8 `form:"numbers"`
}
```

when request a HTTP request,
```url
url://xxx?numbers=[null,2]
```
```go
import 	"github.com/zeromicro/go-zero/rest/httpx"

func ExampleHandler(svcCtx *svc.ServiceContext) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		var req ExampleReq
		if err := httpx.Parse(r, &req); err != nil {
			//...
		}
	}
}

```

 it's panic ,

```prolog
2024-01-12 08:12:05.876	
	caller=internal/log.go:77	trace=bd580c6503f746b3554feb795a47e2c1	span=05ab3e420a415c10

2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/rest/handler/timeouthandler.go:76 +0x333
2024-01-12 08:12:05.876	
created by github.com/zeromicro/go-zero/rest/handler.(*timeoutHandler).ServeHTTP in goroutine 13235
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/rest/handler/timeouthandler.go:82 +0x71
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/rest/handler.(*timeoutHandler).ServeHTTP.func1()
2024-01-12 08:12:05.876	
	/usr/local/go/src/net/http/server.go:2136 +0x29
2024-01-12 08:12:05.876	
net/http.HandlerFunc.ServeHTTP(0xc0005ce280?, {0x27a23b0?, 0xc000d38a80?}, 0x4452dc?)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/rest/handler/recoverhandler.go:21 +0x78
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/rest/handler.RecoverHandler.func1({0x27a23b0?, 0xc000d38a80?}, 0x6f4bbb?)
2024-01-12 08:12:05.876	
	/usr/local/go/src/net/http/server.go:2136 +0x29
2024-01-12 08:12:05.876	
net/http.HandlerFunc.ServeHTTP(0xc000102d00?, {0x27a23b0?, 0xc000d38a80?}, 0x46d7d0?)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/rest/handler/metrichandler.go:21 +0xae
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/rest.(*engine).buildChainWithNativeMiddlewares.MetricHandler.func6.1({0x27a23b0, 0xc000d38a80}, 0x44f445?)
2024-01-12 08:12:05.876	
	/usr/local/go/src/net/http/server.go:2136 +0x29
2024-01-12 08:12:05.876	
net/http.HandlerFunc.ServeHTTP(0xc00009f6c8?, {0x27a23b0?, 0xc000d38a80?}, 0x50?)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/rest/handler/maxbyteshandler.go:24 +0xf8
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/rest.(*engine).buildChainWithNativeMiddlewares.MaxBytesHandler.func8.1({0x27a23b0?, 0xc000d38a80?}, 0x3b0cbe0?)
2024-01-12 08:12:05.876	
	/usr/local/go/src/net/http/server.go:2136 +0x29
2024-01-12 08:12:05.876	
net/http.HandlerFunc.ServeHTTP(0x7a082a98ff7d2e?, {0x27a23b0?, 0xc000d38a80?}, 0xc1603ed5742a1c6a?)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/rest/handler/gunziphandler.go:26 +0x110
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/rest/handler.GunzipHandler.func1({0x27a23b0, 0xc000d38a80}, 0xc000e86400)
2024-01-12 08:12:05.876	
	/usr/local/go/src/net/http/server.go:2136 +0x29
2024-01-12 08:12:05.876	
net/http.HandlerFunc.ServeHTTP(0x0?, {0x27a23b0?, 0xc000d38a80?}, 0xc00009f5d0?)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/rest/handler/authhandler.go:81 +0x3b8
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/rest/handler.Authorize.func1.1({0x27a23b0, 0xc000d38a80}, 0xc000e86400)
2024-01-12 08:12:05.876	
	/usr/local/go/src/net/http/server.go:2136 +0x29
2024-01-12 08:12:05.876	
net/http.HandlerFunc.ServeHTTP(0xc00098fcb0?, {0x27a23b0?, 0xc000d38a80?}, 0xc0007d5fe0?)
2024-01-12 08:12:05.876	
	/code/api/admin/internal/handler/ExampleHandler.go:18 +0x57
2024-01-12 08:12:05.876	
app-server/api/admin/internal/handler.RegisterHandlers.ExampleHandler.func177({0x27a23b0, 0xc000d38a80}, 0xc000e1fd01?)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/rest/httpx/requests.go:43 +0x45
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/rest/httpx.Parse(0xc000e86500, {0x1ea1d20, 0xc000911ec0})
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/rest/httpx/requests.go:76 +0x5a
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/rest/httpx.ParseForm(0xc000100c00?, {0x1ea1d20, 0xc000911ec0})
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/core/mapping/unmarshaler.go:79
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/core/mapping.(*Unmarshaler).Unmarshal(...)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/core/mapping/unmarshaler.go:95 +0x30e
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/core/mapping.(*Unmarshaler).unmarshal(0xc000e86500?, {0x20173a0?, 0xc0007ff110}, {0x1ea1d20, 0xc000911ec0}, {0x0, 0x0})
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/core/mapping/unmarshaler.go:113
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/core/mapping.(*Unmarshaler).unmarshalValuer(...)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/core/mapping/unmarshaler.go:911 +0x3a9
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/core/mapping.(*Unmarshaler).unmarshalWithFullName(0x21ba7c0?, {0x2798c60, 0xc000911f40}, {0x1ea1d20?, 0xc000911ec0}, {0x0, 0x0})
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/core/mapping/unmarshaler.go:540 +0x1c9
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/core/mapping.(*Unmarshaler).processField(0xc000035ca0?, {{0x1e11132, 0x7}, {0x0, 0x0}, {0x27ce5b0, 0x1ecdca0}, {0x1e1113a, 0xe}, 0x8, ...}, ...)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/core/mapping/unmarshaler.go:761 +0x3d6
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/core/mapping.(*Unmarshaler).processNamedField(0xc000035ca0, {{0x1e11132, 0x7}, {0x0, 0x0}, {0x27ce5b0, 0x1ecdca0}, {0x1e1113a, 0xe}, 0x8, ...}, ...)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/core/mapping/unmarshaler.go:791 +0x1d2
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/core/mapping.(*Unmarshaler).processNamedFieldWithValue(0xc000035ca0, {0x27ce5b0, 0x1ecdca0}, {0x1ecdca0?, 0xc000911ec8?, 0x1?}, {{0x1f04840, 0xc0003e0060}, {0x2798c60, 0xc000911fa0}}, ...)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/core/mapping/unmarshaler.go:568 +0x40a
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/core/mapping.(*Unmarshaler).processFieldNotFromString(0xc000911ec8?, {0x27ce5b0, 0x1ecdca0}, {0x1ecdca0?, 0xc000911ec8?, 0x0?}, {{0x1f04840, 0xc0003e0060}, {0x2798c60, 0xc000911fa0}}, ...)
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/core/mapping/unmarshaler.go:241 +0x24a
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/core/mapping.(*Unmarshaler).fillSliceFromString(0x90bb45?, {0x27ce5b0, 0x1ecdca0}, {0x1ecdca0?, 0xc000911ec8?, 0xc00098f468?}, {0x1f04840?, 0xc0003e0060}, {0xc00020d1a8, 0x7})
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/core/mapping/unmarshaler.go:275 +0x435
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/core/mapping.(*Unmarshaler).fillSliceValue(0x2?, {0x1ecdca0?, 0xc0008122d0?, 0x498f5b?}, 0x3b03d60?, 0x23b57e0?, {0x0?, 0x0}, {0xc00020d1a8, 0x7})
2024-01-12 08:12:05.876	
	/usr/local/go/src/runtime/panic.go:914 +0x21f
2024-01-12 08:12:05.876	
panic({0x20486c0?, 0x3aa4180?})
2024-01-12 08:12:05.876	
	/go/pkg/mod/github.com/zeromicro/go-zero@v1.5.6/rest/handler/recoverhandler.go:16 +0x58
2024-01-12 08:12:05.876	
github.com/zeromicro/go-zero/rest/handler.RecoverHandler.func1.1()
2024-01-12 08:12:05.876	
	/usr/local/go/src/runtime/debug/stack.go:24 +0x5e
2024-01-12 08:12:05.876	
runtime/debug.Stack()
2024-01-12 08:12:05.876	
goroutine 13237 [running]:
2024-01-12 08:12:05.876	
2024-01-12T08:12:05.875+08:00	 error 	(/api/example?numbers=[null,2] - 192.168.3.122) runtime error: invalid memory address or nil pointer dereference
```

**Comment by codeErrorSleep** [comment]
url://xxx?numbers=[null,2]
这个传参是不是要改成这种格式的才比较合理的?
url://xxx?numbers[0]=null&numbers[1]=2

**Comment by kevwan** [comment]
Fixed by #3839
